### DISCUSSION TITLE
Improvements for "Multi Tenancy" in CMS

### DISCUSSION THREAD

**Discussion by Michelgelle** [pimcore/pimcore#14746] [Ideas]
In my company, we want to setup  a central pimcore instance for the core pim/dam/mdm Tasks and Create Multiple Websites for our different product-profit-centers.  Every website have it's own brand and style and it is actually quite difficult to separate all the things like templates and so on.

F.E. creating an Areabrick in Pimcore is easy, if the templates are all available under "template/areas", but if we have a structure like "templates/brand1", "templates/brand2"  and a multi-site documents structure all the things like "data object preview" , "template pathes" and so on have to be configured in an advanced way. 
Another point: if one of our advertising-agencies want to create a brand-specific campaign (update one of our profit-center websites), we have to restrict the access for the agency (that works already fine), but there is a need to update the whole Pimcore Instance because of the changes in one brand-specific templates subfolder.

So my Idea is, to have a real "multi-tenancy" Feature for the distribution related parts of pimcore (of course, product data and assets should be managed at the same place), or such things like advanced communication features for a pimcore to pimcore communication, so that one instance could be the mdm/pim instance, and another instance is the cms for only ONE brand / tenant but with data source from the central instance...

**Comment by dpfaffenbauer** [comment]
Try using coreshop/theme-bundle, makes life easier with things like that ;)

**Comment by solverat** [comment]
You may want to checkout our toolbox bundle. We had exact the same issue. Not only that we need different areas for each site but

- we also want to manage areas by using permissions
- use different areas for different sites
- use different themes / frameworks (it is possible to use bootstrap 3 and tailwind in the same pimcore installation)

Bonus: Toolbox improves UX and DX! :)

https://github.com/dachcom-digital/pimcore-toolbox/blob/master/docs/15_Context.md
